farm/data_handler/utils.py

```diff
@@ -214,9 +215,15 @@ def read_dpr_json(file, max_samples=None, proxies=None):
     if not (os.path.exists(file)):
         logger.info(f" Couldn't find {file} locally. Trying to download ...")
         _download_extract_downstream_data(file, proxies=proxies)
-    dicts = json.load(open(file))
-    if max_samples:
-        dicts = random.sample(dicts, min(max_samples, len(dicts)))
+
+    if "jsonl" in file:
```

Comment:
There is a failing test:

```
if "jsonl" in file:
E       TypeError: argument of type 'PosixPath' is not iterable
```
Since file can also be of type PosixPath.
Lets check the file ending with `file.suffix.lower() == ".jsonl"`